kubetest2/internal/deployers/eksapi/nodegroup.go

```diff
@@ -249,6 +249,56 @@ func (m *NodegroupManager) createUnmanagedNodegroupWithEFA(infra *Infrastructure
 	if err != nil {
 		return err
 	}
+	subnetId := infra.subnetsPrivate[0]
+	var capacityReservationId string
+	if opts.CapacityReservation {
+		capacityReservations, err := m.clients.EC2().DescribeCapacityReservations(context.TODO(), &ec2.DescribeCapacityReservationsInput{
+			Filters: []ec2types.Filter{
+				{
+					Name:   aws.String("instance-type"),
+					Values: opts.InstanceTypes,
+				},
+				{
+					Name:   aws.String("state"),
+					Values: []string{"active"},
+				},
+			},
+		})
+		if err != nil {
+			return fmt.Errorf("failed to describe capacity reservation")
+		}
+		var az string
+		for _, cr := range capacityReservations.CapacityReservations {
+			if *cr.AvailableInstanceCount >= int32(opts.Nodes) {
+				capacityReservationId = *cr.CapacityReservationId
+				az = *cr.AvailabilityZone
+				break
+			}
+		}
+		if capacityReservationId == "" {
+			return fmt.Errorf("no capacity reservation found for instance type %s with %d nodes count", opts.InstanceTypes[0], opts.Nodes)
+		}
+		subnet, err := m.clients.EC2().DescribeSubnets(context.TODO(), &ec2.DescribeSubnetsInput{
+			Filters: []ec2types.Filter{
+				{
+					Name:   aws.String("availability-zone"),
+					Values: []string{az},
+				},
+				{
+					Name:   aws.String("subnet-id"),
+					Values: infra.subnetsPrivate,
+				},
+			},
+		})
+		if err != nil {
+			return fmt.Errorf("failed to describe subnet")
+		}
+		if subnet == nil || len(subnet.Subnets) == 0 {
+			return fmt.Errorf("no subnet found for availability zone %s", az)
+		}
+		subnetId = *subnet.Subnets[0].SubnetId
+		klog.Infof("Using capacity reservation: %s", capacityReservationId)
```

Comment:
nit: do you want to also log the subnet Id? or it can be found from other routes?